src/Orleans.Core/Async/UnobservedExceptionsHandler.cs

```diff
@@ -87,5 +76,15 @@ private static void InternalUnobservedTaskExceptionHandler(object sender, Unobse
                 }
             }
         }
+
+        public void Dispose()
+        {
+            lock (lockObject)
```

Comment:
just get rid of the lock field entirely, as it's not really protecting anything anymore